### DISCUSSION TITLE
Is it possible to change the DNS server programatically?

### DISCUSSION THREAD

**Discussion by felipecrs** [vmakeev/huawei_mesh_router#38] [Q&A] [answered]
In this page, I can set the DNS server:

![image](https://user-images.githubusercontent.com/29582865/216130470-f53f4613-0eeb-4665-9a1e-70260586f64b.png)

`192.168.1.10` in the example points to my Home Assistant instance, which in turn is running the AdGuard Home add-on as a DNS server for ad-blocking and malware blocking.

However, I often face an issue: when my HA instance goes down, the internet connection of the whole home goes down as well. Like for example, when I'm restarting HA.

But another situation is when I have a power outage, which is frequent where I live. The UPS which holds my HA instance can hold it up for like 15 minutes, but the UPS which holds my Huawei router can hold it up for like 45 minutes. In other words, in a power outage I only get internet connection for the firsts 15 minutes.

I can of course manually go in the Huawei web UI and change it to something like `1.1.1.1` to restore connectivity, but doing that manually every time is cumbersome.

If there was a way to programmatically set the DNS server of the primary router, I could create an automation in Home Assistant to automatically set the router DNS to `1.1.1.1` and change it back to `192.168.1.10` when power is back. I could do the same thing using an event trigger for when I'm restarting the HA instance.

**Comment by vmakeev** [comment, marked as answer]
Yes, I think it is possible to make a service that will change the connection data

As far as I understand, for this I will need to get the current connection parameters `GET api/ntwk/wan?type=active`, then change the `IPv4DnsServers` field, and send back the request `POST api/ntwk/wan?type=active` with the body of the following type:

```json
{
  "action": "update",
  "data": <previously received by the GET request and modified data>,
  "csrf": {
    "csrf_param": .....,
    "csrf_token": .....
  }
}
```

**Reply by felipecrs** [reply]
Thank you! It's good to hear that it's possible then. But please don't spend time on it as I'm the only one that's asking for this feature so far. In this case, ideally, I should code it myself :)
